R/zoop_downloader.R

```diff
@@ -183,10 +195,135 @@ Zoopdownloader <- function(
+  if("DOP_Meso"%in%Data_sets) {
+
+    DOP_Meso_file<-"DOP_ICF_Mesozooplankton_Abundance2017-2021"
+    DOP_Meso_URL<-paste0(DOP_pkg_url, "/", DOP_entities[DOP_Meso_file])
+
+    DOP_trawls_file<-"DOP_ICF_TowData2017-2021"
+    DOP_trawls_URL<-paste0(DOP_pkg_url, "/", DOP_entities[DOP_trawls_file])
+
+    #download the files
+    if (!file.exists(file.path(Data_folder, DOP_Meso_file)) | Redownload_data) {
+      Tryer(n=3, fun=utils::download.file, url=DOP_Meso_URL,
+            destfile=file.path(Data_folder, DOP_Meso_file), mode="wb", method=Download_method)
+    }
+    if (!file.exists(file.path(Data_folder, DOP_trawls_file)) | Redownload_data) {
+      Tryer(n=3, fun=utils::download.file, url=DOP_trawls_URL,
+            destfile=file.path(Data_folder, DOP_trawls_file), mode="wb", method=Download_method)
+    }
+
+
+    # Import the DOP data
+
+    zoo_DOP_Meso<-readr::read_csv(file.path(Data_folder, DOP_Meso_file))
+    zoo_DOP_trawls<-readr::read_csv(file.path(Data_folder, DOP_trawls_file))
+
+    # Tranform from "wide" to "long" format, add some variables,
+    # alter data to match other datasets
+
+    data.list[["DOP_Meso"]] <- zoo_DOP_Meso %>%
+      tidyr::pivot_longer(cols = !ICF_ID, names_to = "DOP_Meso", values_to = "CPUE") %>%
+      dplyr::left_join(zoo_DOP_trawls) %>%
+      dplyr::mutate(Datetime =  lubridate::ymd(as.character(Date), tz = "America/Los_Angeles") + lubridate::hms(as.character(Start_Time)),
+              Source = "DOP", #add variable for data source
+              SizeClass = "Meso") %>%
+      dplyr::filter(!is.na(.data$Mesozooplankton_Volume)) %>% #get rid of environmental variables with no data
+
+      #Select variables we are interested in. I need to check on the latitude/longitude issue with Sam.
+      dplyr::select(.data$Source, .data$Date, .data$Datetime,
+                    Station = .data$Station_Code, Chl = .data$Chl_a, CondSurf = .data$Conductivity, .data$Secchi, .data$SizeClass,
+                    .data$Temperature, Volume = .data$Mesozooplankton_Volume, BottomDepth = .data$Start_Depth,
+                    .data$DOP_Meso, .data$CPUE, .data$Latitude, .data$Longitude, .data$ICF_ID) %>%
+      dplyr::left_join(Crosswalk %>% #Add in Taxnames, Lifestage, and taxonomic info
+                       dplyr::select(.data$DOP_Meso, .data$Lifestage, .data$Taxname, .data$Phylum,
+                                     .data$Class, .data$Order, .data$Family, .data$Genus, .data$Species)%>% #only retain dop codes
+                         dplyr::filter(!is.na(.data$DOP_Meso))%>% #Only retain Taxnames corresponding to EMP codes
+                         dplyr::distinct(),
+                       by="DOP_Meso")%>%
+      dplyr::filter(!is.na(.data$Taxname),#I might not need this step for this dataset, but just in case
+                    !is.na(CPUE)) %>%  #get rid of the lines with "NA" because the critter wasn't counted in this sample.
```

Comment:
What does this mean? Why wouldn't a critter be counted in a sample? Did they not used a consistent taxonomic list for every sample?

Reply:
THe dataset was origionally in "wide" format. Some of the trawls were all "NA"'s (samples with environmental data, but now zooplankton) and some critteres were only started to be ID'd later, so those entries in earlier samples were "NA"'s for that column. Should I add "DOP_Start" dates to the crosswalk? Let me know if that doesn't make sense, or you want to chat about it real quick.

Reply:
Yeah, if they only started to ID them later then we should add the DOPstart column to the crosswalk. I'm happy to chat about it if you run into issues with that. I think it could be a bit complex how that is implemented. I believe we'd need to make changes to the data-raw script for the crosswalk, `Zoopdownloader`, and `Zoopsynther`